packages/client/hmi-client/src/components/widgets/VegaChart.vue

```diff
@@ -246,6 +246,7 @@ watch(
 		} else {
 			// console.log('render interactive');
 			if (!vegaContainer.value) return;
+			vegaVisualization.value?.finalize(); // dispose previous visualization before creating a new one
```

Comment:
Are there any noticeable impact from calling "finalize()" in this case?

Reply:
I wasn't 100% sure so I just did a test. 

https://github.com/user-attachments/assets/025d3fc7-f130-4c61-9045-47c19279aba9

It seems every time chart re renders, it adds up the memory usage and it doesn't free the memory up.
If you call `finalize()` It seems to free up the memory after some time (I think it's when GC kicks in). 

It also adds up the memory whenever the chart component is mounted (e.g. if you open/close drilldown repeatedly, it adds up quite a bit of memory usage) so I think we also need to explicitly clean up on `unmount` as well..

Reply:
Calling `.finalize()`  before rendering new chart spec and calling it on component unmount seems to solve the issue.